src/V3Undriven.cpp

```diff
@@ -390,6 +391,13 @@ class UndrivenVisitor final : public VNVisitorConst {
                                   << " (IEEE 1364-2005 6.1; Verilog only, legal in SV): "
                                   << nodep->prettyNameQ());
             }
+            if (m_inFTaskRef && nodep->varp()->varType().isNet()) {
```

Comment:
I want to point out that I think it is better if we use `!isProcAssignable` instead of `isNet` here. However, it will give false negative on reg / var ports. Since it seems like port `varType` are all set to "PORT", unless it is annotated with "wire", which it then becomes correctly set to "WIRE". And "PORT" varType is not `isProcAssignable`. However, this is a separate issue that I see here: [#1369](https://github.com/verilator/verilator/issues/1369). Any thoughts?

Reply:
I meant if the assignment is to reg ports, using `!isProcAssignable` will result in an error ("false negative" might be the wrong term there haha). But using `isNet` will work ok. The reason is that reg ports `varType` are not set to "VAR" (for var/reg/logic), it is set to "PORT" which is not in the list of `isProcAssignable`.

Do you agree that `varType` of reg ports should be set to "VAR" instead of "PORT" after parsing? Just like `varType` of wire ports is set to "WIRE"?

If yes, do you think it's okay if I try to make the changes to how `varType` works on ports in this pull request? Or just use `isNet` here for now, and make a separate pull request resolving the `varType` on ports?

Reply:
Ah, that's a "false positive" - meaning it's incorrectly warning.  Using isNet is ok for now.

"PORT" is retained I believe for function I/O.  It could be changed to VAR, but you'll need to clean up whatever fallout - best done in a pre-pull separate from this if you want to - but as above isNet will catch most common misusages.

Reply:
I see, okay, thanks! I'll resolve this one now.